cpp/scripts/run-clang-compile.py

```diff
@@ -0,0 +1,320 @@
+# Copyright (c) 2020-2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+import argparse
+import glob
+import json
+import multiprocessing as mp
+import os
+import re
+import shutil
+import subprocess
+
+
+CLANG_COMPILER = "clang++"
```

Comment:
Just to clarify current status: we won't expect this compile successfully yet, correct? If not, we should add a note to this file, maybe underneath the license, which states clang support is experimental/not fully working yet. In that case, I'm also okay getting the other changes in this PR merged but pushing this script into another PR.

Reply:
Yes exactly, we don't expect successful compilation, at least for some files in the RAFT test suite. I'll add a note there